src/Castle.Windsor.Extensions.MsDependencyInjection/WindsorServiceProviderFactory.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Windsor.Extensions.MsDependencyInjection
+{
+    using System;
+    using System.Collections.Generic;
+    
+    using Castle.MicroKernel;
+    using Castle.Windsor;
+    
+    using Microsoft.Extensions.DependencyInjection;
+
+    public class WindsorServiceProviderFactory : IServiceProviderFactory<IWindsorContainer>, IDisposable
+    {
+        public WindsorServiceProviderFactory()
+        {
+            
+        }
+        public IWindsorContainer CreateBuilder(IServiceCollection services)
+        {
+            var container = services.CreateContainer(this);
+            return container;
+        }
+
+        public IServiceProvider CreateServiceProvider(IWindsorContainer container)
+        {
+            var rootScope = NetCoreRootScope.BeginRootScope();
+            return container.Resolve<IServiceProvider>();
+        }
+
+        public void Dispose()
```

Comment:
Why is this disposable, if there is nothing to dispose?

Reply:
Removed